src/IIIFPresentation/API/Features/Manifest/CanvasPaintingResolver.cs

```diff
@@ -211,13 +223,24 @@ public class CanvasPaintingResolver(
                 StaticWidth = paintedResource.CanvasPainting?.StaticWidth,
                 StaticHeight = paintedResource.CanvasPainting?.StaticHeight
             };
+            
+            if (canvasPaintings.Where(c => c.CanvasOrder != cp.CanvasOrder).Any(c => c.Id == cp.Id))
```

Comment:
Seems odd to do this validation per iteration? Could this be caught up front? If so would it simplify the different `GetCanvasId` methods (previous PR comment)?